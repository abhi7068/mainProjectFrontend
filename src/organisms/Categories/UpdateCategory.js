import React, { Component } from 'react';
import AddCategoryForm from '../../molecules/AddCategoryForm';
import { Card, CardContent, CardHeader } from '@material-ui/core';
import {
  getCategoriesById,
  updateCategoriesById,
} from '../../services/categoryServices';
import { withStyles } from '@material-ui/styles';
import PropTypes from 'prop-types';
import MaterialButton from '../../atoms/MaterialButton';
import { withRouter } from 'react-router-dom';
import SnackBar from '../../atoms/SnackBar';
const styles = {
  root: {
    width: '300px',
    height: '300px',
    margin: '10% auto',
  },
  label: {
    textTransform: 'capitalize',
  },
  display: {
    margin: '5% ',
  },
};

class UpdateCategory extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: {},
      open: false,
      sendUpdateId: this.props.id,
      disabled: false,
    };
    this.onModelChange = this.onModelChange.bind(this);
    this.updateCategory = this.updateCategory.bind(this);
  }

  onModelChange(e) {
    const value = e.target.value;
    this.setState({
      data: { ...this.state.data, title: value },
    });
  }

  componentDidUpdate(prevProps, prevState) {
    //  const { dispatch } = this.props;
    if (prevState.sendUpdateId !== this.state.sendUpdateId) {
      getCategoriesById(this.state.sendUpdateId).then((res) => {
        this.setState({ data: res.data });
      });
    }
  }

  static getDerivedStateFromProps(props, state) {
    //
    if (props.id !== state.sendUpdateId) {
      return { sendUpdateId: this.props.id };
    }
    return null;
  }

  componentDidMount() {
    getCategoriesById(this.state.sendUpdateId).then((res) => {
      this.setState({ data: res.data });
    });
  }

  updateCategory() {
    updateCategoriesById(this.state.sendUpdateId, this.state.data).then(
      this.setState({
        open: true,
        disabled: true,
      })
    );
  }

  render() {
    return (
      <div>
        {this.state.open && (
          <SnackBar path="/admin/listcategories" name="Category Saved" />
        )}
        <Card
          classes={{
            root: this.props.classes.root,
          }}
        >
          <CardHeader title="UPDATE CATEGORY" />

          <CardContent>
            <AddCategoryForm
              onModelChange={this.onModelChange}
              value={this.state.data}
            />
          </CardContent>
          <br />
          <MaterialButton
            data-testid="button"
            disabled={this.state.disabled}
            name="Submit"
            click={this.updateCategory}
          />
        </Card>
      </div>
    );
  }
}
UpdateCategory.propTypes = {
  classes: PropTypes.object,
  root: PropTypes.object,
  display: PropTypes.object,
  id: PropTypes.string,
  history: PropTypes.object,
};

export default withRouter(withStyles(styles)(UpdateCategory));
