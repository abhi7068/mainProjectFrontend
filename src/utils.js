import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import reducer from '../src/store/reducers/rootReducer';
import { Router } from 'react-router-dom';

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const history = createMemoryHistory();
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    );
  }
  Wrapper.propTypes = {
    children: PropTypes.elementType,
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
export * from '@testing-library/react';
export { render };
