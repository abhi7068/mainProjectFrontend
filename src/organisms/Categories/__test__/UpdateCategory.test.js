import UpdateCategory from '../UpdateCategory';
import React from 'react';
import { render } from '../../../utils';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';

it('should render', () => {
  render(<UpdateCategory />);
});
it('should take a snapshot', () => {
  const { asFragment } = render(<UpdateCategory />);
  expect(asFragment(render(<UpdateCategory />))).toMatchSnapshot();
});
it('it renders with correct text', () => {
  const { getByTestId } = render(<UpdateCategory />);
  expect(getByTestId('button')).toHaveTextContent('Submit');
});
it('renders with expected length"', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <UpdateCategory />
    </BrowserRouter>,

    div
  );
  expect(div.querySelectorAll('input')).toHaveLength(1);
});
it('renders with expected length"', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <UpdateCategory />
    </BrowserRouter>,
    div
  );
  expect(div.querySelector('input')).toHaveAttribute('type', 'text');
});
it('renders with expected length"', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <UpdateCategory />
    </BrowserRouter>,
    div
  );
  expect(div.querySelector('input')).toHaveAttribute('name', 'title');
});
