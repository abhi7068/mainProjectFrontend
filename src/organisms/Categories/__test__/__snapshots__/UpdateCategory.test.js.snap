// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiPaper-root MuiCard-root UpdateCategory-root-162 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            UPDATE CATEGORY
          </span>
        </div>
      </div>
      <div
        class="MuiCardContent-root"
      >
        <div>
          <legend
            class="MuiFormLabel-root"
            data-testid="label"
          >
            Title
          </legend>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root"
          style="width: 225px;"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="outlined-basic"
              name="title"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-277 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-278"
                style="width: 0.01px;"
              >
                <span>
                  â€‹
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <br />
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          data-testid="button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Submit
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
